backend/src/supervisors/supervisor.py

```diff
@@ -1,39 +1,39 @@
-import json
-import logging
 from typing import Tuple
 
-from src.utils import get_scratchpad, update_scratchpad
+from src.llm import get_llm
+from src.utils import Config, get_scratchpad, update_scratchpad
 from src.router import get_agent_for_task
 from src.agents import validator_agent
 
-no_tasks_response = "No tasks found to solve"
+no_questions_response = "No questions found to solve"
 unsolvable_response = "I am sorry, but I was unable to find an answer to this task"
 no_agent_response = "I am sorry, but I was unable to find an agent to solve this task"
 
+config = Config()
+router_llm = get_llm(config.router_llm)
```

Comment:
Just wondering, shouldn't the supervisor be llm ignorant?

Reply:
it is a fair point, I'll move it down to the router file itself